src/Frame/Frame.h

```diff
@@ -99,14 +100,23 @@ class Frame {
 
     // Image/Frame info
     casacore::IPosition ImageShape(const StokesSource& stokes_source = StokesSource());
-    size_t Width();     // length of x axis
-    size_t Height();    // length of y axis
-    size_t Depth();     // length of z axis
-    size_t NumStokes(); // if no stokes axis, nstokes=1
-    int CurrentZ();
-    int CurrentStokes();
-    int SpectralAxis();
-    int StokesAxis();
+    casacore::IPosition OriginalImageShape() const; // Image shape from the original file
+    void SetCurrentZ(int z);
+    void SetCurrentStokes(int stokes);
+    void CheckCurrentZ(int& z) const;
+    void CheckCurrentStokes(int& stokes) const;
+    bool IsCurrentChannel(int z, int stokes) const;
+    bool IsCurrentStokes(int stokes) const;
```

Comment:
As discussed in the comment about eliminating the `CURRENT_...` default parameters, the two `Check...` functions should not be necessary, and should be eliminated. After that, the two `IsCurrent...` functions will also no longer be necessary -- the cache functions which use them can directly refer to `CurrentZ()` and `CurrentStokes()` instead.